pkgs/by-name/pr/printrun/package.nix

```
@@ -17,46 +19,38 @@ python3Packages.buildPythonApplication rec {
     hash = "sha256-INJNGAmghoPIiivQp6AV1XmhyIu8SjfKqL8PTpi/tkY=";
   };
 
-  postPatch = ''
-    sed -i -r "s|/usr(/local)?/share/|$out/share/|g" printrun/utils.py
-  '';
-
-  pythonRelaxDeps = [
-    "pyglet"
-    "cairosvg"
-  ];
-
   nativeBuildInputs = [
     glib
     wrapGAppsHook3
   ];
 
-  propagatedBuildInputs = with python3Packages; [
-    appdirs
+  build-system = with python3Packages; [
+    setuptools
     cython
-    dbus-python
-    numpy
-    six
-    wxpython
-    psutil
-    pyglet
-    pyopengl
-    pyserial
-    cffi
-    cairosvg
-    lxml
-    puremagic
   ];
 
+  dependencies =
+    with python3Packages;
+    [
+      pyserial
+      wxpython
+      numpy
+      pyglet
+      psutil
+      lxml
+      platformdirs
+      puremagic
+    ]
+    ++ lib.optional stdenv.hostPlatform.isLinux dbus-python
+    ++ lib.optional stdenv.hostPlatform.isDarwin pyobjc-framework-Cocoa;
+
+  pythonRelaxDeps = [ "pyglet" ];
+
   # pyglet.canvas.xlib.NoSuchDisplayException: Cannot connect to "None"
   doCheck = false;
 
-  setupPyBuildFlags = [ "-i" ];
-
   postInstall = ''
-    for f in $out/share/applications/*.desktop; do
-      sed -i -e "s|/usr/|$out/|g" "$f"
-    done
+    substituteInPlace $out/share/applications/*.desktop --replace /usr $out
```

```suggestion
    substituteInPlace $out/share/applications/*.desktop \
      --replace-fail /usr "$out"
```
but actually desktop files shouldn't contain absolute paths: https://github.com/NixOS/nixpkgs/issues/308324